server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -233,32 +269,57 @@ public void collectSearchShards(String routing, IntConsumer consumer) {
         }
     }
 
-    public static class ExtractFromSource extends IndexRouting {
+    public static class RoutingPathMatching extends IndexRouting {
```

Comment:
Maybe keep the name the same. In both ways we keep extracting routing information from source and this keeps the diff of this change smaller.

Reply:
Indeed, done.